#  -----------------------------     Twitter Sentiment Analysis       --------------------------------


# 1:Import modules

In [1]:
import pandas as pd

## 2: Read and Load the Dataset

In [2]:
file_path = 'Twitter Data/Twitter_Data_1600000.csv'
twitter_data = pd.read_csv(file_path, encoding ='latin')
(row,col)=twitter_data.shape
print(f"Number of rows: {row}\nNumber of columns: {col}")
twitter_data.head()

Number of rows: 1599999
Number of columns: 6


0  1467810369  Mon Apr 06 22:19:45 PDT 2009  NO_QUERY _TheSpecialOne_  \
0  0  1467810672  Mon Apr 06 22:19:49 PDT 2009  NO_QUERY   scotthamilton   
1  0  1467810917  Mon Apr 06 22:19:53 PDT 2009  NO_QUERY        mattycus   
2  0  1467811184  Mon Apr 06 22:19:57 PDT 2009  NO_QUERY         ElleCTF   
3  0  1467811193  Mon Apr 06 22:19:57 PDT 2009  NO_QUERY          Karoli   
4  0  1467811372  Mon Apr 06 22:20:00 PDT 2009  NO_QUERY        joy_wolf   

  @switchfoot http://twitpic.com/2y1zl - Awww, that's a bummer.  You shoulda got David Carr of Third Day to do it. ;D  
0  is upset that he can't update his Facebook by ...                                                                   
1  @Kenichan I dived many times for the ball. Man...                                                                   
2    my whole body feels itchy and like its on fire                                                                    
3  @nationwideclass no, it's not behaving at all....                                                                   
4                      @Kwesidei not the whole crew

##### A problem above dataFrame it not read columns so rows are 1600000 but show 1599999, so defined columns names in below:

In [3]:
# Define columns names
column_names = ['target','ids','date','flag','user','text']
# add columns names to the dataFrame
twitter_data.columns = column_names
# twitter_data = pd.read_csv(file_path, names = column_names, encoding = 'latin')
(row,col) = twitter_data.shape
print(f"Number of rows: {row}\nNumber of columns: {col}")
twitter_data.head()

Number of rows: 1599999
Number of columns: 6


target         ids                          date      flag           user  \
0       0  1467810672  Mon Apr 06 22:19:49 PDT 2009  NO_QUERY  scotthamilton   
1       0  1467810917  Mon Apr 06 22:19:53 PDT 2009  NO_QUERY       mattycus   
2       0  1467811184  Mon Apr 06 22:19:57 PDT 2009  NO_QUERY        ElleCTF   
3       0  1467811193  Mon Apr 06 22:19:57 PDT 2009  NO_QUERY         Karoli   
4       0  1467811372  Mon Apr 06 22:20:00 PDT 2009  NO_QUERY       joy_wolf   

                                                text  
0  is upset that he can't update his Facebook by ...  
1  @Kenichan I dived many times for the ball. Man...  
2    my whole body feels itchy and like its on fire   
3  @nationwideclass no, it's not behaving at all....  
4                      @Kwesidei not the whole crew

## 3: Exploratory Data Analysis(EDA)

### 3.1: Display Basic Information:

In [4]:
twitter_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1599999 entries, 0 to 1599998
Data columns (total 6 columns):
 #   Column  Non-Null Count    Dtype 
---  ------  --------------    ----- 
 0   target  1599999 non-null  int64 
 1   ids     1599999 non-null  int64 
 2   date    1599999 non-null  object
 3   flag    1599999 non-null  object
 4   user    1599999 non-null  object
 5   text    1599999 non-null  object
dtypes: int64(2), object(4)
memory usage: 73.2+ MB


### 3.2: Summary Statatics

In [5]:
twitter_data.describe()


target           ids
count  1.599999e+06  1.599999e+06
mean   2.000001e+00  1.998818e+09
std    2.000001e+00  1.935757e+08
min    0.000000e+00  1.467811e+09
25%    0.000000e+00  1.956916e+09
50%    4.000000e+00  2.002102e+09
75%    4.000000e+00  2.177059e+09
max    4.000000e+00  2.329206e+09

In [6]:
# View First Few Rows
twitter_data.head()

target         ids                          date      flag           user  \
0       0  1467810672  Mon Apr 06 22:19:49 PDT 2009  NO_QUERY  scotthamilton   
1       0  1467810917  Mon Apr 06 22:19:53 PDT 2009  NO_QUERY       mattycus   
2       0  1467811184  Mon Apr 06 22:19:57 PDT 2009  NO_QUERY        ElleCTF   
3       0  1467811193  Mon Apr 06 22:19:57 PDT 2009  NO_QUERY         Karoli   
4       0  1467811372  Mon Apr 06 22:20:00 PDT 2009  NO_QUERY       joy_wolf   

                                                text  
0  is upset that he can't update his Facebook by ...  
1  @Kenichan I dived many times for the ball. Man...  
2    my whole body feels itchy and like its on fire   
3  @nationwideclass no, it's not behaving at all....  
4                      @Kwesidei not the whole crew

In [7]:
# Check the missing Values
twitter_data.isnull().sum()

target    0
ids       0
date      0
flag      0
user      0
text      0
dtype: int64

In [8]:
# Unique values
twitter_data.nunique()

target          2
ids       1598314
date       774362
flag            1
user       659775
text      1581465
dtype: int64

In [9]:
# Value Count
twitter_data['target'].value_counts()

target
4    800000
0    799999
Name: count, dtype: int64

In [10]:
twitter_data.columns

Index(['target', 'ids', 'date', 'flag', 'user', 'text'], dtype='object')

In [11]:
twitter_data.shape

(1599999, 6)

In [12]:
twitter_data.count()

target    1599999
ids       1599999
date      1599999
flag      1599999
user      1599999
text      1599999
dtype: int64

In [13]:
twitter_data['text'].isnull().sum()

0